code/logmessage.py

```diff
@@ -82,30 +98,28 @@ def full_message(self):
     def log_message_serial_fields(cls):
         if cls._log_message_serial_fields_cache:
             return cls._log_message_serial_fields_cache
-        cache = {}
-        subclasses = cls.mro()
-        subclasses.append(cls)
-        for subcls in reversed(subclasses):
-            try:
-                fields = subcls._log_message_serial_fields
-                # Enable short hands when there is a 1:1 between constructor argument,
-                # serial format, and the field name
-                if isinstance(fields, list):
-                    cache.update({x: x for x in fields})
-                else:
-                    cache.update(fields)
-            except AttributeError:
-                pass
+        cache = merge_fields_on_subclasses(cls, "_log_message_serial_fields")
         cls._log_message_serial_fields_cache = cache
         assert 'log_id' not in cache, "The log_id field is reserved for internal usage"
         return cache
 
+    @classmethod
+    def log_message_serial_converters(cls):
+        if cls._log_message_serial_converters_cache:
+            return cls._log_message_serial_converters_cache
+        cache = merge_fields_on_subclasses(cls, "_log_message_serial_converters")
+        cls._log_message_serial_converters_cache = cache
+        assert 'log_id' not in cache, "The log_id field is reserved for internal usage"
```

Comment:
Note that log_ids are currently not converted.  I am ok with that; just wanted to make it explicit.